File: Builder/HtmlBuilder/WithBuilder/HtmlElement.cs
using System.Text;

namespace WithBuilder
{
    public class HtmlElement
    {
        public string Tag, Value;
        public List<HtmlElement> Elements = new List<HtmlElement>();

        private const int identSize = 2;

        public HtmlElement()
        {

        }

        public HtmlElement(string tag, string value)
        {
            Tag = tag;
            Value = value;
        }

        private string ToStringImpl(int indent)
        {
            var sb = new StringBuilder();
            var i = new string(' ', identSize * indent);
            sb.AppendLine($"{i}<{Tag}>");

            if (!string.IsNullOrWhiteSpace(Value))
            {
                sb.Append(new string(' ', identSize * indent + 1));
                sb.AppendLine(Value);
            }

            foreach (var e in Elements)
            {
                sb.Append(e.ToStringImpl(indent + 1));
            }
            sb.AppendLine($"{i}</{Tag}>");

            return sb.ToString();
        }

        public override string ToString()
        {
            return ToStringImpl(0);
        }
    }
}

File: Builder/Python/HtmlBuilder/WithBuilder/html_element.py
class HtmlElement:
    indent_size: int = 2

    def __init__(self, name: str = "", text: str = ""):
        self.name: str = name
        self.text: str = text
        self.elements: list = []

    def __str(self, indent: int) -> str:
        lines: list = []
        i = ' ' * (indent * self.indent_size)
        lines.append(f'{i}<{self.name}>')

        if self.text:
            i1 = ' ' * ((indent + 1) * self.indent_size)
            lines.append(f'{i1}{self.text}')

        for e in self.elements:
            lines.append(e.__str(indent + 1))

        lines.append(f'{i}</{self.name}>')
        return '\n'.join(lines)

    def __str__(self) -> str:
        return self.__str(0)

    # @staticmethod
    # def create(name):
    #     return HtmlBuilder(name)
